src/org/rascalmpl/dap/RascalDebugAdapter.java

```diff
@@ -506,48 +521,221 @@ public CompletableFuture<Void> pause(PauseArguments args) {
+                response.setResult("Evaluation not possible when program is not suspended");
+                return response;
+            }
 
-                case "variables": // Called from the "variables" view when copying the value of a variable (and maybe in other situations?)
-                    // In this case the expression already contains the value of the variable. We just return it.
-                    response.setResult(expr);
-                    break;
+            EvalResult er = evaluateExpression(expr, frameId);
 
-                default:
-                    break;
+            if (er.output instanceof IWebContentOutput) {
+                try {
+                    var webContent = (IWebContentOutput) er.output;
+                    services.browse(webContent.webUri(), webContent.webTitle(), webContent.webviewColumn());
+                    response.setResult("Web Content displayed");
+                    return response;
+                } catch (UnsupportedOperationException _ignored) { }
             }
 
+            if (er.result != null) { // Successful evaluation
+                // IRascalResult doesn't expose isVoid; check value==null as proxy for void
+                if (er.result.getValue() == null) {
+                    response.setResult("void");
+                    response.setType("void");
+                }
+                else {
+                    RascalVariable tempVar = new RascalVariable(
+                        er.result.getValue().getType(),
+                        "",
+                        er.result.getValue(),
+                        services);
+                    if(tempVar.hasSubFields()){
+                        suspendedState.addVariable(tempVar);
+                        VariableSubElementsCounter counter = er.result.getValue().accept(new VariableSubElementsCounterVisitor());
+                        tempVar.setIndexedVariables(counter.getIndexedVariables());
+                        tempVar.setNamedVariables(counter.getNamedVariables());
+                    }
+
+                    response.setResult(tempVar.getDisplayValue());
+                    response.setType(er.result.getValue().getType().toString());
+                    response.setVariablesReference(tempVar.getReferenceID());
+                    response.setNamedVariables(tempVar.getNamedVariables());
+                    response.setIndexedVariables(tempVar.getIndexedVariables());
+                }
+            }
+            else if (er.output != null) { // Evaluation resulted in error with output
+                // Render the ICommandOutput to plain text for the DAP client
+                var sw = new StringWriter();
+                var pw = new PrintWriter(sw, true);
+                try {
+                    er.output.asPlain().write(pw, true);
+                } catch (Exception _e) {
+                    // fallback
+                    pw.print(er.output.toString());
```

Comment:
what if the `PrintWriter` throw the exception? I think it would be better to write this directly to the `StringWriter`.